bokehjs/test/unit/models/glyphs/patches.ts

```diff
@@ -0,0 +1,43 @@
+import {expect} from "chai"
+
+import {create_glyph_view, set_scales} from "./glyph_utils"
+import {Patches, PatchesView} from "@bokehjs/models/glyphs/patches"
+import {Geometry} from "@bokehjs/core/geometry"
+
+describe("Glyph (using  as a concrete Glyph)", () => {
+
+  describe("GlyphView", () => {
+    it("should hit test rects for containment", async () => {
+      const data = {xs: [[0, 10, 5], [5, 10, 10, 5]], ys: [[0, 0, 10], [10, 10, 20, 20]]}
+      const glyph = new Patches({
+        xs: {field: "xs"},
+        ys: {field: "ys"}
+      })
+
+      const glyph_view = (await create_glyph_view(glyph, data) as PatchesView)
+      set_scales(glyph_view, "linear")
+      glyph_view.map_data()
+
+      const geometry1: Geometry = { type: "rect", sx0: -1, sy0: 201, sx1: 21, sy1: 179}
+      const geometry2: Geometry = { type: "rect", sx0: 1,  sy0: 200, sx1: 20, sy1: 180}
+      const geometry3: Geometry = { type: "rect", sx0: 9,  sy0: 181, sx1: 21, sy1: 159}
+      const geometry4: Geometry = { type: "rect", sx0: 10, sy0: 180, sx1: 19, sy1: 165}
+      const geometry5: Geometry = { type: "rect", sx0: 5,  sy0: 190, sx1: 15, sy1: 170}
+      const geometry6: Geometry = { type: "rect", sx0: -1, sy0: 201, sx1: 21, sy1: 159}
+
+      const result1 = glyph_view.hit_test(geometry1)!
+      const result2 = glyph_view.hit_test(geometry2)!
+      const result3 = glyph_view.hit_test(geometry3)!
+      const result4 = glyph_view.hit_test(geometry4)!
+      const result5 = glyph_view.hit_test(geometry5)!
+      const result6 = glyph_view.hit_test(geometry6)!
+
+      expect(result1.indices).to.be.deep.equal([0])
```

Comment:
A better approach than using `!` operator to get rid of `null`, is to use type assertion functions, like this:
```diff
diff --git a/bokehjs/test/unit/models/glyphs/patches.ts b/bokehjs/test/unit/models/glyphs/patches.ts
index 5fa30c4e2..31ee9804b 100644
--- a/bokehjs/test/unit/models/glyphs/patches.ts
+++ b/bokehjs/test/unit/models/glyphs/patches.ts
@@ -3,6 +3,7 @@ import {expect} from "chai"
 import {create_glyph_view, set_scales} from "./glyph_utils"
 import {Patches, PatchesView} from "@bokehjs/models/glyphs/patches"
 import {Geometry} from "@bokehjs/core/geometry"
+import {assert} from "@bokehjs/core/util/assert"
 
 describe("Glyph (using  as a concrete Glyph)", () => {
 
@@ -25,13 +26,14 @@ describe("Glyph (using  as a concrete Glyph)", () => {
       const geometry5: Geometry = { type: "rect", sx0: 5,  sy0: 190, sx1: 15, sy1: 170}
       const geometry6: Geometry = { type: "rect", sx0: -1, sy0: 201, sx1: 21, sy1: 159}
 
-      const result1 = glyph_view.hit_test(geometry1)!
+      const result1 = glyph_view.hit_test(geometry1)
       const result2 = glyph_view.hit_test(geometry2)!
       const result3 = glyph_view.hit_test(geometry3)!
       const result4 = glyph_view.hit_test(geometry4)!
       const result5 = glyph_view.hit_test(geometry5)!
       const result6 = glyph_view.hit_test(geometry6)!
 
+      assert(result1 != null)
       expect(result1.indices).to.be.deep.equal([0])
       expect(result2.indices).to.be.deep.equal([])
       expect(result3.indices).to.be.deep.equal([1])
```
This uses a TypeScript 3.7 feature call assertion types and allows ensure behavior both at type level and at run time. After the call to `assert()`, `result1` is of type `Selection` instead of `Selection | null`.

Reply:
I still need to figure out how to use chai's expect API instead of plain `assert()`, for better error reporting.

Reply:
Thanks, Was just cargo-culting the equivalent tests for another glyph.